transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -233,6 +234,8 @@ internal class MessagingClientImpl(
                 if (stateMachine.isConnected()) {
                     shouldConfigureAfterConnect = false
                     stateMachine.onError(code, message)
+                } else {
+                    eventHandler.onEvent(ErrorEvent(errorCode = code, message = message))
```

Comment:
Under which scenarios could we fall into this scenario? Trying to understand how we signal an `ErrorEvent` (a type of `StructuredMessageEvent`) here instead of a state error.

Reply:
Mostly regular 4xx error range. For instance: 
Toggle off the typing indicator in Admin console and try sending a `userTyping` request will return Forbidden 403.
In that case there are no reason to fall into error state, but rather notify UI with event.

Reply:
That makes sense to me, that response errors that are due to the client might not cause us to fall into the error state but instead send an error event. I guess what I'm wondering is that right now it's written so that if a `ClientReponseError` happens while connected state then we go to error state, otherwise we report `ErrorEvent`. Would we want all response errors in the client range to report `ErrorEvent`? e.g.

```
…
is ErrorCode.ClientResponseError
-> eventHandler.onEvent(ErrorEvent(errorCode = code, message = message))
is ErrorCode.ServerResponseError,
is ErrorCode.RedirectResponseError,
-> {
    if (stateMachine.isConnected()) {
        shouldConfigureAfterConnect = false
        stateMachine.onError(code, message)
    }
}
…
```

Reply:
The ClientResponseError during `State.Connected` is an edge case. It points out to that `configureSession` request has failed, and therefore we can not proceed with any action. This is why I transition MessengerClient into the State.Error.

Reply:
Oh! I was confusing `isConnected` with `isConfigured`, so that explains why I was having a hard time wrapping my head around it.